examples/step-89/step-89.cc

```diff
@@ -0,0 +1,1884 @@
+        non_matching_faces,
+        pressure_r,
+        velocity_r,
+        c_r,
+        rho_r,
+        nm_mapping_info);
+
+    // Compute the the maximum speed of sound, needed for the computation of
+    // the time-step size.
+    double speed_of_sound_max = 0.0;
+    for (const auto &mat : materials)
+      speed_of_sound_max = std::max(speed_of_sound_max, mat.second.first);
+
+    ConditionalOStream pcout(
+      std::cout, (Utilities::MPI::this_mpi_process(MPI_COMM_WORLD) == 0));
+
+    // Setup time integrator.
+    RungeKutta2<dim, Number, remote_value_type> time_integrator(
+      inverse_mass_operator, acoustic_operator);
+
+    // Run time loop with Courant number 0.1.
+    time_integrator.run(matrix_free,
+                        /*Cr*/ 0.2,
+                        end_time,
+                        speed_of_sound_max,
+                        initial_condition,
+                        vtk_prefix);
+  }
+} // namespace Step89
+
+
+// @sect3{Driver}
+//
+// Finally, the driver executes the different versions of handling non-matching
+// interfaces.
+int main(int argc, char *argv[])
+{
+  using namespace dealii;
+  constexpr int dim = 2;
+  using Number      = double;
+
+  Utilities::MPI::MPI_InitFinalize mpi(argc, argv, 1);
+  std::cout.precision(5);
+  ConditionalOStream pcout(std::cout,
+                           (Utilities::MPI::this_mpi_process(MPI_COMM_WORLD) ==
+                            0));
+
+  const unsigned int refinements = 2;
+  const unsigned int degree      = 5;
+
+  // Construct non-matching triangulation and fill non-matching boundary IDs.
+  parallel::distributed::Triangulation<dim> tria(MPI_COMM_WORLD);
+
+  pcout << "Create non-matching grid..." << std::endl;
+
+  std::set<types::boundary_id> non_matching_faces;
+  Step89::build_non_matching_triangulation(tria,
+                                           non_matching_faces,
+                                           refinements);
+
+  pcout << " - Refinement level: " << refinements << std::endl;
+  pcout << " - Number of cells: " << tria.n_cells() << std::endl;
+
+  // Setup MatrixFree.
+
+  pcout << "Create DoFHandler..." << std::endl;
+  DoFHandler<dim> dof_handler(tria);
+  dof_handler.distribute_dofs(FESystem<dim>(FE_DGQ<dim>(degree), dim + 1));
+  pcout << " - Number of DoFs: " << dof_handler.n_dofs() << std::endl;
+
+  AffineConstraints<Number> constraints;
+  constraints.close();
+
+  pcout << "Setup MatrixFree..." << std::endl;
+  typename MatrixFree<dim, Number>::AdditionalData data;
+  data.mapping_update_flags = update_gradients | update_values;
+  data.mapping_update_flags_inner_faces =
+    update_quadrature_points | update_values;
+  data.mapping_update_flags_boundary_faces =
+    data.mapping_update_flags_inner_faces;
+
+  MatrixFree<dim, Number> matrix_free;
+  matrix_free.reinit(
+    MappingQ1<dim>(), dof_handler, constraints, QGauss<dim>(degree + 1), data);
+
+
+  pcout << "Run vibrating membrane testcase..." << std::endl;
```

Comment:
Could we create a new sub(sub)section?

Reply:
[6c93294](https://github.com/dealii/dealii/pull/16299/commits/6c93294e6da113c2c19c0670ea25ce3a232714cc)